src/components/GraphiQL/GraphiQL.tsx

```diff
@@ -0,0 +1,480 @@
+                    {modifier}
+                    {' + '}
+                    <code className="graphiql-key">Enter</code>
+                  </td>
+                  <td>Execute query</td>
+                </tr>
+                <tr>
+                  <td>
+                    <code className="graphiql-key">Ctrl</code>
+                    {' + '}
+                    <code className="graphiql-key">Shift</code>
+                    {' + '}
+                    <code className="graphiql-key">P</code>
+                  </td>
+                  <td>Prettify editors</td>
+                </tr>
+                <tr>
+                  <td>
+                    <code className="graphiql-key">Ctrl</code>
+                    {' + '}
+                    <code className="graphiql-key">Shift</code>
+                    {' + '}
+                    <code className="graphiql-key">M</code>
+                  </td>
+                  <td>Merge fragments definitions into operation definition</td>
+                </tr>
+                <tr>
+                  <td>
+                    <code className="graphiql-key">Ctrl</code>
+                    {' + '}
+                    <code className="graphiql-key">Shift</code>
+                    {' + '}
+                    <code className="graphiql-key">C</code>
+                  </td>
+                  <td>Copy query</td>
+                </tr>
+                <tr>
+                  <td>
+                    <code className="graphiql-key">Ctrl</code>
+                    {' + '}
+                    <code className="graphiql-key">Shift</code>
+                    {' + '}
+                    <code className="graphiql-key">R</code>
+                  </td>
+                  <td>Re-fetch schema using introspection</td>
+                </tr>
+              </tbody>
+            </table>
+            <p>
+              The editors use{' '}
+              <a
+                href="https://codemirror.net/5/doc/manual.html#keymaps"
+                target="_blank"
+                rel="noopener noreferrer"
+              >
+                CodeMirror Key Maps
+              </a>{' '}
+              that add more short keys. This instance of Graph<em>i</em>QL uses{' '}
+              <code>{props.keyMap || 'sublime'}</code>.
+            </p>
+          </div>
+        </div>
+      </Dialog> */}
+    </div>
+  );
+}
+
+// Configure the UI by providing this Component as a child of GraphiQL.
+function GraphiQLToolbar<TProps>(props: PropsWithChildren<TProps>) {
+  return <>{props.children}</>;
+}
+
+GraphiQLToolbar.displayName = 'GraphiQLToolbar';
+
+// Determines if the React child is of the same type of the provided React component
+function isChildComponentType<T extends ComponentType>(
+  child: any,
+  component: T,
+): child is T {
+  if (
+    child?.type?.displayName &&
+    child.type.displayName === component.displayName
+  ) {
+    return true;
+  }
+
+  return child.type === component;
```

Comment:
**suggestion**: These function should be in a seperate utils file. In general this component is pretty large, almost 500 lines. It would be good idea to try to split it into smaller files.

Reply:
@Droniu This component is un-packed from the official `graphiql` package. It easier to keep it as-is for reference. Eventually, I'd refactor it as you suggest.